aiohttp/connector.py

```diff
@@ -464,11 +491,24 @@ def _create_connection(self, req):
 
         for hinfo in hosts:
             try:
-                return (yield from self._loop.create_connection(
-                    self._factory, hinfo['host'], hinfo['port'],
+                host = hinfo['host']
+                port = hinfo['port']
+                conn = yield from self._loop.create_connection(
+                    self._factory, host, port,
                     ssl=sslcontext, family=hinfo['family'],
                     proto=hinfo['proto'], flags=hinfo['flags'],
-                    server_hostname=hinfo['hostname'] if sslcontext else None))
+                    server_hostname=hinfo['hostname'] if sslcontext else None)
+                if self._verify_fingerprint:
+                    sock = conn[0]._sock
+                    # gives DER-encoded cert as a sequence of bytes (or None)
+                    cert = sock.getpeercert(binary_form=True)
```

Comment:
`conn` is `(transport, protocol)` pair.
For getting socket object from transport please call `transport.get_extra_info('socket')`.
Also you have skip certificate check for non-ssl socket (`request.ssl` is `False`).